navigator-ui: Alpha Release

Manoj-nathwani opened the pull request.

Manoj-nathwani commented:
We now have tests!

```
$ yarn test
yarn run v1.22.17
$ jest
 PASS  lib/buttonLogic.test.js
  The "Task Complete" button should
    If the task is the last task in the breadcrumbs
      and is set to manual
        ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsCompleted onclick action (3 ms)
        ✓ and is completed, should return an enabled checked button with the markTaskAsIncomplete onclick action (1 ms)
      and is set to  not manual
        ✓ and is incomplete, should return an enabled unchecked button with onClick=null
        ✓ and is completed, should return an enabled checked button with onClick=null
    If the task is not the last task in the breadcrumbs
      and is set to manual
        ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsCompleted onclick action (1 ms)
        ✓ and is completed, should return an enabled checked button with the markTaskAsIncomplete onclick action (1 ms)
      and is set to  not manual
        ✓ and is incomplete, should return an enabled unchecked button with onClick=null
        ✓ and is completed, should return an enabled checked button with onClick=null
  The "Prior Task" button should
    If the task is set to manual
      ✓ and if breadcrumbs length > 1, should return an enabled button with the stepCurrentTaskBackOnce onclick action
      ✓ and if breadcrumbs length <= 1, should return a disabled button (1 ms)
    If the task is not set to manual
      ✓ and if breadcrumbs length > 1, should return an enabled button with the stepCurrentTaskBackOnce onclick action (1 ms)
      ✓ and if breadcrumbs length <= 1, should return a disabled button (1 ms)
  The "Next Task" button should
    ✓ If the task is the last task in the breadcrumbs, should return an disabled button (1 ms)
    ✓ If the task is not the last task in the breadcrumbs, should return an enabled button with the onclick stepCurrentTaskForwardOnce action
  The "What's Next" button should
    ✓ If the task is completed, should return an enabled button with the onclick fetchLatestWorkflowState action
    ✓ If the task is incomplete, should return an enabled button with the onclick stepCurrentTaskForwardOnce and fetchLatestWorkflowState actions (6 ms)

Test Suites: 1 passed, 1 total
Tests:       16 passed, 16 total
Snapshots:   0 total
Time:        0.652 s, estimated 1 s
Ran all test suites.
Done in 2.07s.
```

jonathansberry commented:
that looks good to me! 

...providing that "stepCurrentTaskForwardOnce" means "skip this current task" 

It just seems an odd name for the action?  Since you mark the current task for skip and it could step you forward 10 steps.  I'd suggest using a number like "skipCurrentTask" might be better and more consistent with the rest of the system? But I don't know your code well so may there is a reason for this?

Manoj-nathwani commented on lib/buttonLogic.js:
```diff
@@ -0,0 +1,92 @@
+export const actions = {
+  markTaskAsCompleted: 'markTaskAsCompleted',
+  markTaskAsIncomplete: 'markTaskAsIncomplete',
+  stepCurrentTaskBackOnce: 'stepCurrentTaskBackOnce',
+  stepCurrentTaskForwardOnce: 'stepCurrentTaskForwardOnce',
+  fetchLatestWorkflowState: 'fetchLatestWorkflowState'
```
`markTaskAsCompleted` endpoint

`markTaskAsIncomplete` endpoint

`stepCurrentTaskBackOnce` go through the breadcrumbs and fetch the previous task details

`stepCurrentTaskForwardOnce` go through the breadcrumbs and fetch the next task details

add `skipTask` endpoint

Manoj-nathwani replied:
Change the last test to calling the `skipTask` action

Manoj-nathwani replied:
rename `stepCurrentTaskBackOnce` to `getPreviousTask`
rename `stepCurrentTaskForwardOnce ` to `getNextTask`
add `skipTask`
TODO: figure out unskipping a task

Manoj-nathwani commented:
```
$ yarn test
yarn run v1.22.17
$ jest
 PASS  lib/buttonLogic.test.js
  The "Task Complete" button should
    If the task is the last task in the breadcrumbs
      and is set to manual
        ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsCompleted onclick action (4 ms)
        ✓ and is completed, should return an enabled checked button with the markTaskAsIncomplete onclick action
      and is set to  not manual
        ✓ and is incomplete, should return an enabled unchecked button with onClick=null (1 ms)
        ✓ and is completed, should return an enabled checked button with onClick=null (1 ms)
    If the task is not the last task in the breadcrumbs
      and is set to manual
        ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsCompleted onclick action (1 ms)
        ✓ and is completed, should return an enabled checked button with the markTaskAsIncomplete onclick action (1 ms)
      and is set to  not manual
        ✓ and is incomplete, should return an enabled unchecked button with onClick=null (1 ms)
        ✓ and is completed, should return an enabled checked button with onClick=null
  The "Prior Task" button should
    If the task is set to manual
      ✓ and if breadcrumbs length > 1, should return an enabled button with the getPreviousTask onclick action
      ✓ and if breadcrumbs length <= 1, should return a disabled button
    If the task is not set to manual
      ✓ and if breadcrumbs length > 1, should return an enabled button with the getPreviousTask onclick action
      ✓ and if breadcrumbs length <= 1, should return a disabled button (1 ms)
  The "Next Task" button should
    ✓ If the task is the last task in the breadcrumbs, should return an disabled button (1 ms)
    ✓ If the task is not the last task in the breadcrumbs, should return an enabled button with the onclick getNextTask action
  The "What's Next" button should
    ✓ If the task is completed, should return an enabled button with the onclick [fetchLatestWorkflowState] actions (1 ms)
    ✓ If the task is incomplete, should return an enabled button with the onclick [getNextTask, fetchLatestWorkflowState] actions (1 ms)

Test Suites: 1 passed, 1 total
Tests:       16 passed, 16 total
Snapshots:   0 total
Time:        0.709 s, estimated 1 s
Ran all test suites.
Done in 2.15s.
```

- Renamed `stepCurrentTaskBackOnce` to `getPreviousTask`
- Renamed `stepCurrentTaskForwardOnce` to `getNextTask`
- Added `getNextTask` action
- See changes to the last two tests

Manoj-nathwani commented:
```
yarn test
yarn run v1.22.17
$ jest
 PASS  lib/actionButtons.test.js
  The "Task Complete" button should
    If the task isLatestTask in the breadcrumbs
      and is set to manual
        ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsComplete onclick action (3 ms)
        ✓ and is complete, should return an enabled checked button with the markTaskAsIncomplete onclick action (1 ms)
      and is set to not manual
        ✓ and is incomplete, should return an enabled unchecked button with the toggleCompleteStateLocally onclick action (1 ms)
    If the task !isLatestTask in the breadcrumbs
      and is set to manual
        ✓ and is incomplete, should return an enabled unchecked button with the markTaskAsComplete onclick action (1 ms)
        ✓ and is complete, should return an enabled checked button with the markTaskAsIncomplete onclick action (1 ms)
      and is set to not manual
        ✓ and is incomplete, should return an disabled unchecked button (1 ms)
        ✓ and is complete, should return an disabled checked button (1 ms)
  The "Prior Task" button should
    If the task is set to manual
      ✓ and if !isOldestTask in the breadcrumbs, should return an enabled button with the getPreviousTask onclick action (1 ms)
      ✓ and if isOldestTask in the breadcrumbs, should return a disabled button
    If the task is not set to manual
      ✓ and if breadcrumbs length > 1, should return an enabled button with the getPreviousTask onclick action
      ✓ and if breadcrumbs length <= 1, should return a disabled button
  The "Next Task" button should
    ✓ If the task isLatestTask in the breadcrumbs, should return an disabled button
    ✓ If the task !isLatestTask in the breadcrumbs, should return an enabled button with the onclick getNextTask action (1 ms)
  The "What's Next" button should
    ✓ If the task is complete, should return an enabled button with the onclick [fetchLatestWorkflowState] actions (1 ms)
    ✓ If the task is incomplete, should return an enabled button with the onclick skipTaskAndFetchLatestWorkflowState actions (1 ms)

Test Suites: 1 passed, 1 total
Tests:       15 passed, 15 total
Snapshots:   0 total
Time:        0.984 s, estimated 1 s
Ran all test suites.
Done in 2.81s.
```
